app/javascript/flavours/glitch/features/featured/index.jsx

```diff
@@ -26,145 +29,110 @@ const messages = defineMessages({
-    hidden: getAccountHidden(state, accountId),
-  };
-};
-
-class Featured extends ImmutablePureComponent {
-
-  static propTypes = {
-    params: PropTypes.shape({
-      acct: PropTypes.string,
-      id: PropTypes.string,
-    }).isRequired,
-    dispatch: PropTypes.func.isRequired,
-    accountId: PropTypes.string,
-    acct: PropTypes.string,
-    remote: PropTypes.bool,
-    remoteUrl: PropTypes.string,
-    multiColumn: PropTypes.bool,
-    intl: PropTypes.object.isRequired,
-    isAccount: PropTypes.bool,
-    isLoading: PropTypes.bool,
-    suspended: PropTypes.bool,
-    hidden: PropTypes.bool,
-    featuredTags: ImmutablePropTypes.list,
-  };
-
-  _load () {
-    const { accountId, isAccount, dispatch } = this.props;
-
-    if (!isAccount) dispatch(fetchAccount(accountId));
-    dispatch(fetchFeaturedTags(accountId));
-  }
-
-  componentDidMount () {
-    const { params: { acct }, accountId, dispatch } = this.props;
-
-    if (accountId) {
-      this._load();
+const Featured = ({ params, multiColumn, type }) => {
+  const dispatch = useAppDispatch();
+  const intl = useIntl();
+  const columnRef = useRef(null);
+  const showTags = type === 'tags';
+
+  const accountId = useAppSelector((state) => params.id || state.getIn(['accounts_map', normalizeForLookup(params.acct)]));
+  const acct = useAppSelector((state) => state.getIn(['accounts', accountId, 'acct']));
+  const remoteUrl = useAppSelector((state) => state.getIn(['accounts', accountId, 'url']));
+  const isAccount = useAppSelector((state) => !!(state.getIn(['accounts', accountId])));
+  const suspended = useAppSelector((state) => state.getIn(['accounts', accountId, 'suspended'], false));
+  const hidden = useAppSelector((state) => getAccountHidden(state, accountId));
+  // Featured tags are still returned for limited accounts, preventing the limited account hint from showing, so explicitely set an empty list
+  const featuredItems = useAppSelector((state) => !hidden ? state.getIn(['user_lists', showTags ? 'featured_tags' : 'featured_accounts', accountId, 'items'], ImmutableList()) : ImmutableList());
+  const isLoading = useAppSelector((state) => !accountId || state.getIn(['user_lists', showTags ? 'featured_tags' : 'featured_accounts', accountId, 'isLoading'], true));
+  const hasMore = useAppSelector((state) => !accountId || !!state.getIn(['user_lists', showTags ? 'featured_tags' : 'featured_accounts', accountId, 'next']));
+
+  const handleHeaderClick = useCallback(() => columnRef.current?.scrollTop(), []);
+
+  const handleLoadMore = debounce(() => {
+    dispatch(expandPinnedAccounts(accountId));
+  }, 300, { leading: true });
+
+  useEffect(() => {
+    if (!accountId) {
+      dispatch(lookupAccount(params.acct));
     } else {
-      dispatch(lookupAccount(acct));
+      if (!isAccount) dispatch(fetchAccount(accountId));
+      dispatch(fetchFeaturedTags(accountId));
+      dispatch(fetchPinnedAccounts(accountId, 10));
```

Comment:
The limit should match `DEFAULT_ACCOUNTS_LIMIT` or be at least higher, as this is so low that pretty much immediately more items get loaded in. This was mostly for testing anyway as creating accounts is tedious even with a script. Though the tedious part was following and pinning them.
Might be worth rewriting the underlying action.